### ISSUE TITLE
Veros land columns publish 273.15 K as SST; the land-fill substitution never fires

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-esm#127]
## What is wrong

`VerosComponent._derived_fields` (in `jem/components/veros_component.py`, introduced by PR #125) substitutes a land-fill value when extracting the published sea surface temperature:

```python
# Land columns carry a fill value rather than a temperature; replace
# them with a plausible constant so downstream components never see
# an unphysical SST through the mask.
sea_surface_temperature = jnp.where(
    sea_surface_temperature < 100, 288.15, sea_surface_temperature)
```

For both shipped setups that branch is **dead**, and the comment's premise is wrong. `double_drake.py:224` and `earth.py:334` both build the cold start as `... * cold_start_temperature_celsius * vs.maskT`, so a land column holds `0.0` degC, not a large negative sentinel. After the `+ 273.15` offset it is exactly `273.15`, which is not `< 100`, so the substitution never runs and land cells publish `273.15 K` as their SST.

Measured on the packaged setups (PR #125): ocean cells publish a uniform `288.0336 K`, land cells `273.15 K`, both before and after that PR's initialization fix — so this is pre-existing behaviour, unchanged by it, and consistent between `initialize()` and `step()`.

## Why it probably has not bitten yet

The atmosphere applies its own land-sea mask, so the value published over land is not expected to be read as a temperature there. It is nonetheless wrong in two ways worth resolving deliberately rather than by accident:

1. `273.15 K` is above `mask_fluxes_under_ice`'s `271.35 K` threshold (`jem/fluxes.py`), so land columns read as "ice-free" in that test.
2. The stated intent — "downstream components never see an unphysical SST" — is not actually achieved, and the next person to read the comment will believe it is.

## What needs deciding

What a land column *should* publish is a design question, not a mechanical fix, which is why this is an issue rather than part of PR #125:

- keep a plausible constant, but detect land from `maskT` rather than from a magnitude threshold that assumes a sentinel these setups do not use;
- publish `NaN` and require every consumer to mask (loudest, but changes the exchange's contract and interacts with the double-`where` AD pattern this repo already relies on); or
- drop the substitution entirely and document that the field is only meaningful under the ocean mask.

The masking convention has to stay identical between `initialize()` and `step()` — they now share `_derived_fields`, so that part is already structural.

## Where the hooks are

- `jem/components/veros_component.py::VerosComponent._derived_fields` — the substitution and its comment.
- `jem/components/veros/setups/double_drake.py:224`, `jem/components/veros/setups/earth.py:334` — the `* vs.maskT` cold start that makes land `0` degC.
- `jem/fluxes.py::mask_fluxes_under_ice` — the `271.35 K` threshold that land currently passes.

Found while fixing the round-16 review finding on PR #125 (`initialize()` publishing a placeholder SST); cross-referenced from the code comment.
